src/main/java/com/amplitude/Amplitude.java

```diff
@@ -215,6 +215,29 @@ public synchronized void flushEvents() {
     }
   }
 
+  /**
+   * Check the status of the client, return true if any of the following situation:
+   * 1. Events sit in memory waiting for flush are more than flush threshold
+   * 2. Events in retry buffer are more than 16,000
+   * 3. When setRecordThrottledId(true), the userId or deviceId of the input event was throttled and retry attempt for the userId/deviceId not finished.
+   * Can be called before logEvent(event) and add some wait time if return true.
+   *
+   * @param event The event to be sent.
+   * @return true if client is busy or event may be throttled.
+   */
+  public boolean shouldWait(Event event) {
+    return eventsToSend.size() > eventUploadThreshold || httpTransport.shouldWait(event);
```

Comment:
can we add some tests for the new method?

Reply:
Will the first part of logic happen? We can keep it just wonder if that would happen.

Reply:
Test added. The first part may happen for future multithreading handling change.